src/dc_imex.rs

```diff
@@ -799,172 +717,151 @@ unsafe fn export_backup(context: &Context, dir: *const libc::c_char) -> bool {
         dc_delete_file(context, &dest_path_filename);
     }
 
-    success
+    if success {
+        Ok(())
+    } else {
+        bail!("failed");
+    }
```

Comment:
you can just write `ensure!(success, "failed")`

Reply:
done